urlfc1c29991b_nil_mish_InJava_tgz-pJ8-edu_usc_cs_nsl_indexing_test_TestTrieProtoJ8/src/edu/usc/cs/nsl/counting/InputGenerator.java

```diff
@@ -0,0 +1,75 @@
+package edu.usc.cs.nsl.counting;
+
+import java.security.*;
+import java.util.Arrays;
+import java.util.Random;
+import java.io.*;
+import java.math.BigInteger;
+
+/**
+ * The real challenge is to generate an input good enough for this problem.
+ * Text files are not good enough as English words are over in 100,000 words.
+ * Here we will generate a dictionary by counting numbers and taking hash of them.
+ * We can then randomly create additional strings from these hashes by randomly 
+ * splitting it. 
+ * @author nilmish
+ *
+ */
+public class InputGenerator {
+
+	
+	/**
+	 * @param args
+	 * @throws IOException 
+	 * @throws NoSuchAlgorithmException 
+	 */	
+	public static void main(String[] args) throws IOException, NoSuchAlgorithmException {
+		MessageDigest md = MessageDigest.getInstance("sha1");
+		int end_value = Integer.parseInt(args[0]);
+		BufferedWriter bw = new BufferedWriter(new FileWriter(args[1]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *bw*. The resource is referenced at lines 29, 35, 38. The resource is closed at line 41. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *bw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).